doc/intro.md

```diff
@@ -291,66 +286,62 @@ concept: _cut-through_. With this addition, a MimbleWimble chain gains:
+We can say the following is true for any valid transaction (ignoring fees for simplicity):
 
     sum(outputs) - sum(inputs) = kernel_excess
 
-The same holds true for blocks themselves once we realize a block is simply a set of aggregated inputs, outputs and transaction kernels. We can sum the tx outputs, subtract the sum of the tx inputs and compare the resulting Pedersen commitment to the sum of the kernel excesses -
+The same holds true for blocks themselves once we realize a block is simply a set of aggregated inputs, outputs and transaction kernels. We can sum the outputs, subtract the inputs from it and equating the resulting Pedersen commitment to the sum of the kernel excesses:
 
     sum(outputs) - sum(inputs) = sum(kernel_excess)
 
 Simplifying slightly, (again ignoring transaction fees) we can say that MimbleWimble blocks can be treated exactly as MimbleWimble transactions.
 
 ##### Kernel Offsets
 
-There is a subtle problem with MimbleWimble blocks and transactions as described above. It is possible (and in some cases trivial) to reconstruct the constituent transactions in a block. This is clearly bad for privacy. This is the "subset" problem - given a set of inputs, outputs and transaction kernels a subset of these will recombine to  reconstruct a valid transaction.
+There is a subtle problem with MimbleWimble blocks and transactions as described above. It is possible (and in some cases trivial) to reconstruct the constituent transactions in a block. This is clearly bad for privacy. This is the "subset" problem: given a set of inputs, outputs, and transaction kernels a subset of these will recombine to reconstruct a valid transaction.
 
-For example, given the following two transactions -
+Consider the following two transactions:
 
     (in1, in2) -> (out1), (kern1)
     (in3) -> (out2), (kern2)
 
-We can aggregate them into the following block (or aggregate transaction) -
+We can aggregate them into the following block (or aggregate transaction):
 
     (in1, in2, in3) -> (out1, out2), (kern1, kern2)
 
-It is trivially easy to try all possible permutations to recover one of the transactions (where it sums successfully to zero) -
+It is trivially easy to try all possible permutations to recover one of the transactions (where it successfully sums to zero):
 
     (in1, in2) -> (out1), (kern1)
 
-We also know that everything remaining can be used to reconstruct the other valid transaction -
+We also know that everything remaining can be used to reconstruct the other valid transaction:
 
     (in3) -> (out2), (kern2)
 
-To mitigate this we include a _kernel offset_ with every transaction kernel. This is a blinding factor (private key) that needs to be added back to the kernel excess to verify the commitments sum to zero -
+To mitigate this we include a _kernel offset_ with every transaction kernel. This is a public key that needs to be added to the kernel excess to ensure the commitments sum to zero:
```

Comment:
The offset is a blinding factor here.
If the offset was a commitment it would be trivial to build a tx with any arbitrary offset and any corresponding outputs (ask me how I know).

Reply:
It says 
>This is a blinding factor (private key) that needs to be added back to the kernel excess to verify the commitments sum to zero.

If the offset is a blinding factor (scalar), then surely you can't add it to a curve point (the kernel excess is `(r2-r1)*G`). At the time of writing I realize that it's probably meant to say *excess value* and not *kernel excess*, which would make the equation
>`sum(outputs) - sum(inputs) = sum(kernel_excess) + kernel_offset*G`

rather than the currently written

>`sum(outputs) - sum(inputs) = sum(kernel_excess) + kernel_offset`

If I understand this correctly, the kernel excess is actually not `r*G` where *r* is the sum of the input and output blinding factors, but rather `(r-offset)*G`, and the offset (not-so-private key) is published so that the miners can validate the individual transactions during block creation.

I'll edit shortly if there are no objections (i.e. if I'm not wrong still).

Reply:
> ```sum(outputs) - sum(inputs) = kernel_excess + kernel_offset*G```

Yes I think this is correct.

I think we were trying to balance accuracy with simplicity here in the intro. Once you start bringing both the excess and the offset into the equation it gets significantly more complex to explain clearly.